nomos-services/mempool/src/lib.rs

```diff
@@ -153,8 +153,16 @@ where
                 Some(msg) = service_state.inbound_relay.recv() => {
                     match msg {
                         MempoolMsg::Add { item, key, reply_channel } => {
-                            match pool.add_item(key, item) {
+                            match pool.add_item(key, item.clone()) {
                                 Ok(_id) => {
+                                    // Broadcast the transaction to the network
```

Comment:
```suggestion
                                    // Broadcast the item to the network
```
Probably better to change like this? since mempool is generic now.